src/exodus/test/sp_tests.cpp

```diff
@@ -0,0 +1,102 @@
+#include "../sp.h"
+
+#include "test/test_bitcoin.h"
+#include "main.h"
+#include "miner.h"
+#include "wallet/wallet.h"
+
+#include <boost/test/unit_test.hpp>
+
+struct SpTestingSetup : public TestingSetup
```

Comment:
There is already `TestChain100Setup` in `src/test/test_bitcoin.h` that provides block generation function. The problems with that class is it will generating 100 blocks immediately. My idea is we should move block generating function to `TestingSetup` and add one method to generate empty blocks there.

Reply:
hmm I'm ok to generate 100 blocks. it's not hurt performance much.